paasta_tools/kubernetes_tools.py

```diff
@@ -2595,6 +2630,13 @@ def write_annotation_for_kubernetes_service(
         )
 
 
+def list_all_matching_deployments(kube_client: KubeClient) -> Sequence[KubeDeployment]:
```

Comment:
Should we call this `list_all_paasta_deployments`? It's unclear what `matching` means. Might also be worth adding a docstring.